### PR TITLE
Split on numbers

### PR DESCRIPTION
**Pull request by FreeApophis**
Numbers were not handled so far
* Decision: numbers are treated as words in PascalCase or camelCase

### CODE REVIEW

File: Messerli.ChangeCase/Messerli.ChangeCase.csproj [hunk 1]
```diff
@@ -6,7 +6,7 @@
         <Product>Messerli.ChangeCase</Product>
         <Description>Transform a string between different casings.</Description>
         <PackageTags>Utility Casing</PackageTags>
-        <Version>1.0.0</Version>
+        <Version>1.1.</Version>
```
Comment (tautropfli) [inline Messerli.ChangeCase/Messerli.ChangeCase.csproj]: Something's wrong with this version 👀 

See the [failing build](https://github.com/messerli-informatik-ag/change-case/pull/3/checks?check_run_id=3229779168): 
```
/home/runner/.dotnet/sdk/5.0.100/NuGet.targets(131,5): error : '1.1.' is not a valid version string. (Parameter 'value') [/home/runner/work/change-case/change-case/Messerli.ChangeCase.sln]
```

File: Messerli.ChangeCase.Test/FirstCharacterToLowerCaseTest.cs [hunk 2]
```diff
@@ -12,6 +12,10 @@ public class FirstCharacterToLowerCaseTest
         [InlineData("this_is_a_long_snake_case_name", "this_is_a_long_snake_case_name")]
         [InlineData("HTML", "hTML")]
         [InlineData("htmlDocument", "htmlDocument")]
+        [InlineData("end_2_end", "end_2_end")]
+        [InlineData("BUSINESS_2_BUSINESS", "bUSINESS_2_BUSINESS")]
+        [InlineData("End2End", "end2End")]
+        [InlineData("business2Business", "business2Business")]
```
Comment (tautropfli) [inline Messerli.ChangeCase.Test/FirstCharacterToLowerCaseTest.cs:18]: There are no tests for numbers that consist of more than one digit. Is that intentional?

File: Messerli.ChangeCase/StringCaseExtensions.SplitResult.cs [hunk 3]
```diff
@@ -11,13 +11,13 @@ public static partial class StringCaseExtensions
             public readonly int NextStartIndex;
 
             public SplitResult(int nextStartIndex, Option<string> result = default)
-            {
-                Result = result
+                => (Result, NextStartIndex) = (FlattenResult(result), nextStartIndex);
+
+            private static string FlattenResult(Option<string> result)
```
Comment (tautropfli) [inline Messerli.ChangeCase/StringCaseExtensions.SplitResult.cs]: Unrelated nitpick (feel free to resolve): `result` is only none when used as the seed for a `Sequence.Generate`, right? Wouldn't it make sense to just pass in `string.Empty` from there and make `result` non-optional?

**Review by tautropfli — CHANGES_REQUESTED** [review]

File: Messerli.ChangeCase/Messerli.ChangeCase.csproj (hunk 1 above)
Comment (FreeApophis) [reply Messerli.ChangeCase/Messerli.ChangeCase.csproj]: Fixed in 9686cb6702f9458507fda80324387aa24877c4fe

File: Messerli.ChangeCase.Test/FirstCharacterToLowerCaseTest.cs (hunk 2 above)
Comment (FreeApophis) [reply Messerli.ChangeCase.Test/FirstCharacterToLowerCaseTest.cs:18]: Added and implemented in b6a105f61574ed19e5a2af14554013d4a6a941da

File: Messerli.ChangeCase/StringCaseExtensions.SplitResult.cs (hunk 3 above)
Comment (FreeApophis) [reply Messerli.ChangeCase/StringCaseExtensions.SplitResult.cs]: Addressed in 0d227abcfa9d64e525f701a020b8153b9e0df991

**Review by tautropfli — APPROVED** [review]

**Pull request merged**